lib/python/src/bailo/helper/release.py

```diff
@@ -148,11 +157,61 @@ def download(self, filename: str, write: bool = True, path: str | None = None) -
         if write:
             if path is None:
                 path = filename
-            with open(path, "wb") as f:
-                f.write(res.content)
+            total_size = int(res.headers.get("content-length", 0))
+            block_size = 1024
+
+            if no_color == 1:
+                colour = "white"
+            else:
+                colour = "green"
+
+            with tqdm(
+                total=total_size,
+                unit="B",
+                unit_scale=True,
+                unit_divisor=block_size,
+                postfix=f"downloading {filename} as {path}",
+                colour=colour,
+            ) as t:
+                with open(path, "wb") as f:
+                    for data in res.iter_content(block_size):
+                        t.update(len(data))
+                        f.write(data)
 
         return res
 
+    def download_all(self, path: str = os.getcwd(), include: Union[list, str] = None, exclude: Union[list, str] = None):
```

Comment:
Old an new type annotations are being used here instead we should use `list | str | None`

Reply:
Done